### PR TITLE
Feature/task 22507 switch to sliding lifecycle policy

### PR DESCRIPTION
**Pull request by aceltuk**

### CODE REVIEW

File: polaris-pipeline/coordinator/Functions/Orchestration/Client/Case/CaseClient.cs [hunk 1]
```diff
@@ -111,83 +78,24 @@ public class CaseClient
...
-                        }
-                        catch (Exception)
-                        {
-                            _telemetryClient.TrackEventFailure(telemetryEvent);
-                            throw;
-                        }
-
+                        return await _orchestrationProvider.DeleteCaseAsync(orchestrationClient, currentCorrelationId, caseIdNum);
 
                     case "PUT":
```
Comment (Alastair-Pandelus) [inline polaris-pipeline/coordinator/Functions/Orchestration/Client/Case/CaseClient.cs:97]: Might be good to remove the PUT. It was used in testing to set the tracker, but that isn't possible now it's created from multiple source enties

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+            return new HttpResponseMessage(success ? HttpStatusCode.OK : HttpStatusCode.InternalServerError);
+        }
+        catch (Exception)
+        {
+            _telemetryClient.TrackEventFailure(telemetryEvent);
+            throw;
+        }
+    }
+
+    public async Task<HttpResponseMessage> UpdateTrackerAsync(IDurableOrchestrationClient orchestrationClient, Guid correlationId, string caseId, CaseOrchestrationPayload casePayload, HttpRequestMessage req)
```
Comment (Alastair-Pandelus) [inline polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs:179]: Might be good to remove this method now. It was used in testing to support the manual setting of the tracker from Postman, but that isn't possible now it's created from multiple source entities

File: polaris-pipeline/coordinator/Domain/Dto/DurableInstanceRecord.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using Newtonsoft.Json;
+
+namespace coordinator.Domain.Dto;
+
+public class DurableInstanceRecord
```
Comment (Alastair-Pandelus) [inline polaris-pipeline/coordinator/Domain/Dto/DurableInstanceRecord.cs:5]: I've been using the convention of NameDto for Dto Objects, see Common\Dto\**\*

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+using Common.Configuration;
+using Common.Constants;
+using Common.Services.BlobStorageService.Contracts;
+using coordinator.Domain.Dto;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace coordinator.Providers;
+
+public class OrchestrationProvider : IOrchestrationProvider
```
Comment (Alastair-Pandelus) [inline polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs:29]: This looks better hived off

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+            var response = await httpClient.SendAsync(request);
+            response.EnsureSuccessStatusCode();
+            
+            var jsonString = await response.Content.ReadAsStringAsync();
+            if (string.IsNullOrWhiteSpace(jsonString))
+                return caseId;
+            
+            var results =  JsonConvert.DeserializeObject<List<DurableInstanceRecord>>(jsonString);
+            var targetInstance = results.FirstOrDefault(i => i.Name == "RefreshCaseOrchestrator" && i.RuntimeStatus is "Completed" or "Failed");
```
[the remark this answers is not in the document]
Comment (aceltuk) [reply polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs]: Fair enough. Done.

File: polaris-pipeline/coordinator/Functions/Orchestration/Client/Case/CaseClient.cs (hunk 1 above)
Comment (aceltuk) [reply polaris-pipeline/coordinator/Functions/Orchestration/Client/Case/CaseClient.cs]: Done

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs (hunk 2 above)
Comment (aceltuk) [reply polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs]: Done

File: polaris-pipeline/coordinator/Domain/Dto/DurableInstanceRecord.cs (hunk 3 above)
Comment (aceltuk) [reply polaris-pipeline/coordinator/Domain/Dto/DurableInstanceRecord.cs]: Ah, makes sense. Done.

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs (hunk 4 above)
Comment (aceltuk) [reply polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs:30]: Cool :-)

**Review by Alastair-Pandelus — APPROVED** [review]

File: polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Alastair-Pandelus) [reply polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs]: "Completed" or "Failed" still there?

**Review by Alastair-Pandelus — DISMISSED** [review]

**Pull request merged**